services/order.py

```diff
@@ -0,0 +1,29 @@
+from django.contrib.auth import get_user_model
+
+from db.models import Order, Ticket, MovieSession
+from django.db import transaction
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict], username: str, date: str | None = None
```

Comment:
Consider using `datetime` objects for `date` instead of strings to ensure proper handling of date and time data. This will also help avoid potential issues with date formatting and time zone handling.

Reply:
Agree